packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -965,6 +967,16 @@ export class ContainerRuntime extends TypedEventEmitter<IContainerRuntimeEvents>
             metadata,
         );
 
+        const shouldRunExpiry = this.runtimeOptions.gcOptions?.gcAllowed
+        && !this.runtimeOptions.gcOptions?.disableGC
+        && !this.runtimeOptions.gcOptions?.runSweep;
+
+        if(shouldRunExpiry) {
+            const defaultContainerRuntimeExpiryMs = 30 * 24 * 60 * 60 * 1000;
+            const setReadOnlyCall = () => this.setReadOnly();
```

Comment:
I think I provided this feedback somewhere else, but it can't be just read-only state, as there is absolutly no indication to the user what just happened. I believe it should be critical error that should close container with very specific error code. This will also get recorded properly in telemetry for us to track it over time

Reply:
Also note that this is pretty big memory hog (will keep all of the runtime around) even if container is closed for other reasons. I think we need to cancel that timer when container runtime is disposed for any reason.